pkg/controllermanager/controller/shoot/shoot_maintenance_control.go

```diff
@@ -224,3 +219,81 @@ func hasMaintainNowAnnotation(shoot *gardenv1beta1.Shoot) bool {
 	operation, ok := shoot.Annotations[common.ShootOperation]
 	return ok && operation == common.ShootOperationMaintain
 }
+
+// MaintainMachineImage determines if a shoots machine image has to be maintained and in case returns the target image
+func MaintainMachineImage(shoot *gardenv1beta1.Shoot, cloudProfile *gardenv1beta1.CloudProfile, shootCurrentImage *gardenv1beta1.MachineImage) (bool, *gardenv1beta1.MachineImage, error) {
+	machineImagesFound, machineImages, err := helper.DetermineMachineImages(*cloudProfile, shootCurrentImage.Name)
+	if err != nil {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: %s", err.Error())
+	}
+	if !machineImagesFound || len(machineImages) == 0 {
+		return false, nil, fmt.Errorf("failure while determining the machine images in the CloudProfile: no machineImage with name '%s' (specified in shoot) could be found in the cloud cloudProfile '%s'", shootCurrentImage.Name, cloudProfile.Name)
+	}
+
+	// checks if forceUpdate Time & applicable (has the specified version)
+	forceUpdateRequired, forceUpdateTime := ForceUpdateRequired(shootCurrentImage, machineImages)
+
+	// safely dereference due to defaulting in api server
+	autoUpdateOfMachineImageConfigured := *shoot.Spec.Maintenance.AutoUpdate.MachineImageVersion
+	if forceUpdateRequired || autoUpdateOfMachineImageConfigured {
+		versionUpdatePossible, updateMachineImage, err := VersionUpdatePossible(shootCurrentImage, machineImages)
+		if err != nil {
+			return false, nil, nil
```

Comment:
The returned error shouldn't be ignored since it functionally affects the the auto upgrade feature.